weekly-scan/scheduler.yaml

```diff
@@ -0,0 +1,76 @@
+apiVersion: "v1"
+kind: "Template"
+metadata:
+  name: weekly-scan-scheduler-template
+labels:
+  type: "scheduler"
+  app: "weekly-scan"
+objects:
+  - kind: ConfigMap
+    apiVersion: v1
+    metadata:
+      name: weekly-scan-scheduler-script
+    data:
+      script: |-
+        GOT_ANY='0'
+        OS_TOKEN=`cat /tmp/jenkins-secret/token`
+        oc login ${OS_MASTER} --insecure-skip-tls-verify=true --token=${OS_TOKEN};
+        for item in `oc get bc -o name --selector app=weekly-scan --selector type=pipeline`; do
+            GOT_ANY='1'
```

Comment:
Keeping this inside `for` loop would cause `GOT_ANY`'s value to be set to `1` every time. I think we want it to be `1` for checking things in the end. Can we instead have an `if` condition before triggering the `for` loop that sets value to `1`. That way we can exit before the `for` loop. Example:

```bash
GOT_ANY='0'
OS_TOKEN=`cat /tmp/jenkins-secret/token`
oc login ${OS_MASTER} --insecure-skip-tls-verify=true --token=${OS_TOKEN};

if oc get bc -o name --selector app=weekly-scan --selector type=pipeline | wc -l > 1; then
  GOT_ANY=1
else
  echo "No projects to trigger"
  exit 0;
fi

for item in `oc get bc -o name --selector app=weekly-scan --selector type=pipeline`; do
  oc start-build ${item};
  sleep ${BATCH_DELAY}
done

Reply:
Going a step further, we can do things without the extra variable as well:

```bash
OS_TOKEN=`cat /tmp/jenkins-secret/token`
oc login ${OS_MASTER} --insecure-skip-tls-verify=true --token=${OS_TOKEN};

if oc get bc -o name --selector app=weekly-scan --selector type=pipeline | wc -l ==0; then
  echo "No projects to trigger"
  exit 0;
fi

for item in `oc get bc -o name --selector app=weekly-scan --selector type=pipeline`; do
  oc start-build ${item};
  sleep ${BATCH_DELAY}
done
```

Reply:
Hmm, makes sens. Implementing